src/v0/destinations/lemnisk/transform.js

```diff
@@ -0,0 +1,133 @@
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  DIAPI_CONFIG_CATEGORIES,
+  DIAPI_MAPPING_CONFIG
+} = require("./config");
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  simpleProcessRouterDest
+} = require("../../util");
+const { fetchPlatform } = require("./utils");
+const {
+  ConfigurationError,
+  TransformationError,
+  InstrumentationError
+} = require("../../util/errorTypes");
+
+const responseBuilder = (message, category, destination, platform) => {
+  let payload;
+  const { Config } = destination;
+  const response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  if (platform === "pl") {
+    const { plWriteKey, pl } = Config;
+    if (!plWriteKey || !plWriteKey.length > 0 || !pl || !pl.length > 0) {
+      throw new ConfigurationError(
+        "Configuration for Web Mode requires write key and region url"
+      );
+    }
+    payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+    payload.writeKey = plWriteKey;
+    payload.context.userAgent = {
+      ua: payload.context.userAgent
+    };
+    response.endpoint = pl;
+    response.headers = {
+      "Content-Type": "application/json"
+    };
+  } else {
+    // diapi
+    const { diapi, apiKey, passKey, diapiWriteKey, srcId } = Config;
+    if (
+      !diapi ||
+      !diapi.length > 0 ||
+      !apiKey ||
+      !apiKey.length > 0 ||
+      !passKey ||
+      !passKey.length > 0
+    ) {
+      throw new ConfigurationError(
+        "Configuration for Server Mode requires Api key, Pass Key and region url"
+      );
+    }
+    payload = constructPayload(message, DIAPI_MAPPING_CONFIG[category.name]);
+    if (diapiWriteKey !== "") {
```

Comment:
Can we employ util functions for these check instead of explicit string comparison? Sort of secures us in case of any edge cases

Reply:
`isdefinedandnotnull` this one right ?

Reply:
Done